tutorials/shaders/shader_reference/shader_functions.rst

```diff
@@ -0,0 +1,3804 @@
+    They can each refer to any of several actual types.
+
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | alias           | actual types                                        | glsl documentation alias |
+    +=================+=====================================================+==========================+
+    | vec_type        | float, vec2, vec3, or vec4                          | genType                  |
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | vec_int_type    | int, ivec2, ivec3, or ivec4                         | genIType                 |
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | vec_uint_type   | uint, uvec2, uvec3, or uvec4                        | genUType                 |
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | vec_bool_type   | bool, bvec2, bvec3, or bvec4                        | genBType                 |
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | mat_type        | mat2, mat3, or mat4                                 | mat                      |
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | gvec4_type      | vec4, ivec4, or uvec4                               | gvec4                    |
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | gsampler2D      | sampler2D, isampler2D, or uSampler2D                | gsampler2D               |
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | gsampler2DArray | sampler2DArray, isampler2DArray, or uSampler2DArray | gsampler2DArray          |
+    +-----------------+-----------------------------------------------------+--------------------------+
+    | gsampler3D      | sampler3D, isampler3D, or uSampler3D                | gsampler3D               |
+    +-----------------+-----------------------------------------------------+--------------------------+
+
+    If  any of these are specified for multiple parameters, they must all be the same type unless otherwise noted.
+
+.. _shading_componentwise:
+
+.. note::
+    Many functions that accept one or more vectors or matrices perform the described function on each component of the vector/matrix.
+    Some examples:
+
+    +-----------------------------------+-----------------------------------------------------+
+    | Operation                         | Equivalent Scalar Operation                         |
+    +===================================+=====================================================+
+    | ``sqrt(vec2(4,64))``              | ``vec2(sqrt(4),sqrt(64))``                          |
```

Comment:
```suggestion
    | ``sqrt(vec2(4, 64))``             | ``vec2(sqrt(4), sqrt(64))``                         |
```
For all these

Reply:
```suggestion
    | ``sqrt(vec2(4.0, 64.0))``              | ``vec2(sqrt(4.0), sqrt(64.0))``                          |
```
Across all the shader pages, floats are styled with the explicit decimal to distinguish from ints.